src/draw.c

```diff
@@ -120,6 +125,40 @@ static struct color string_to_color(const char *str)
         return hex_to_color(0xF, 1);
 }
 
+static struct gradient string_to_gradient(char **strv)
+{
+        struct gradient g;
+        g.length = 0;
+
+        if (strv == NULL) {
+                LOG_W("Invalid empty color gradient string");
+                g.cs = g_malloc(sizeof(struct color));
+                g.cs[g.length++] = hex_to_color(0xF, 1);
+                /* return black (only one color) on error */
+        } else {
+                g.cs = g_malloc0(g_strv_length(strv) * sizeof(struct color));
+                for (int i = 0; strv[i] != NULL; i++) {
+                        char *end, *str = strv[i];
+                        if (STR_EMPTY(str)) continue;
+
+                        uint_fast32_t val = strtoul(str+1, &end, 16);
```

Comment:
Why `uint_fast32_t` ... shouldn't it be `unsigned long`?

Are you using `str+1` here to skip the initial `#`? This should be done, iff it is actually a `#`.

How about handling of named colors like `Yellow`? Maybe I'm missing something here, but I couldn't find, where this is handled.

Reply:
Again, i am following the previous behavior (but I don't like it either)